redisinsight/api/src/modules/rdi/rdi-pipeline.service.spec.ts

```diff
@@ -0,0 +1,403 @@
+      providers: [
+        RdiPipelineService,
+        {
+          provide: RdiClientProvider,
+          useFactory: mockRdiClientProvider,
+        },
+        {
+          provide: RdiPipelineAnalytics,
+          useValue: {
+            sendRdiPipelineFetched: jest.fn(),
+            sendRdiPipelineFetchFailed: jest.fn(),
+            sendRdiPipelineDeployed: jest.fn(),
+            sendRdiPipelineDeployFailed: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    service = module.get(RdiPipelineService);
+    rdiClientProvider = module.get(RdiClientProvider);
+    analytics = module.get(RdiPipelineAnalytics);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('getSchema', () => {
+    it('should call getSchema on the RdiClientProvider and return the result', async () => {
+      const schema = { schema: {} };
+      const rdiClient = generateMockRdiClient(rdiClientMetadata);
+      rdiClientProvider.getOrCreate.mockResolvedValueOnce(rdiClient);
+      jest.spyOn(rdiClient, 'getSchema').mockResolvedValueOnce(schema);
+
+      const result = await service.getSchema(rdiClientMetadata);
+
+      expect(rdiClientProvider.getOrCreate).toHaveBeenCalledWith(rdiClientMetadata);
+      expect(rdiClient.getSchema).toHaveBeenCalled();
+      expect(result).toEqual(schema);
+    });
+  });
+
+  describe('getPipeline', () => {
+    it('should call getPipeline on the RdiClientProvider and return the result', async () => {
+      const pipeline = { pipeline: {} };
+      rdiClientProvider.getOrCreate.mockResolvedValue({
+        getPipeline: jest.fn().mockResolvedValue(pipeline),
+      });
+
+      const result = await service.getPipeline(rdiClientMetadata);
+
+      expect(rdiClientProvider.getOrCreate).toHaveBeenCalledWith(rdiClientMetadata);
+      expect(result).toEqual(pipeline);
+    });
+
+    it('should call sendRdiPipelineFetched on the RdiPipelineAnalytics if successful', async () => {
+      const pipeline = { pipeline: {} };
+      rdiClientProvider.getOrCreate.mockResolvedValue({
+        getPipeline: jest.fn().mockResolvedValue(pipeline),
+      });
+
+      await service.getPipeline(rdiClientMetadata);
+
+      expect(analytics.sendRdiPipelineFetched).toHaveBeenCalledWith(rdiClientMetadata.id, pipeline);
+    });
+
+    it('should call sendRdiPipelineFetchFailed on the RdiPipelineAnalytics and throw an error if unsuccessful',
+      async () => {
+        const error = new Error('Failed to get pipeline');
+        jest.spyOn(rdiClientProvider, 'getOrCreate').mockRejectedValue(error);
+
+        await expect(service.getPipeline(rdiClientMetadata)).rejects.toThrow(wrapHttpError(error));
+        expect(analytics.sendRdiPipelineFetchFailed).toHaveBeenCalledWith(error, rdiClientMetadata.id);
+      });
+  });
+
+  describe('dryRunJob', () => {
+    it('should call getOrCreate on rdiClientProvider with the correct metadata', async () => {
+      const client = {
```

Comment:
This will work but is it better to create single mockedClient (which already exists as I remember) and use `mockClient.someMethod.mockResolvedValueOnce` everywhere without mocking everything inside tests